### PR TITLE
Archive resource history

### LINKED ISSUE #2491: Resource history blows up document size
**Describe the bug**

Resource history is saved indefinitely along side the current resource version. This causes the document to get larger and larger overtime but CosmosDB is limited to 2MB per item.

This is happening in our CICD environment when many runs change the firewall share service until it gets too large and the API starts to output errors (from cosmos) about the document size. As a temp measure I deleted some items from the history object.

Probably best to think of a way to address this internally (either save last X versions, archive older history items elsewhere, etc.).

There was an agreement about going on and crearing a new container for archiveing older history items.
The tasks that are planned are:

- [ ] Create a new container for history
- [ ] Create history item in a new container
- [ ] Change history item schema to include also parent id 
- [ ] Change the query (but still return it as part of the resource as a list) 
- [ ] Create an API to return only history for UI usage and undo the list in the resource schema 
- [ ] Fix unit tests 
- [ ] Write migration script

### PR DESCRIPTION
**Pull request by anatbal**
# Fixes #2491
## What is being addressed

- Creation of a new container name ResourceHistory, with partition key of "resourceId"
- Creation of ResourceHistoryRepository class contains query history for resourceId  and create history for resourceId
- Save each history item in the new container
- Migration script to transform the old model to the new one
- Add a new route to get history for each resource
- Change the way UI fetch history
Note - These changes are not touching airlock just yet

### CODE REVIEW

**Review by yuvalyaron — APPROVED** [review]
nice work!
added a comment for fixing a typo

File: api_app/db/repositories/resources_history.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    @staticmethod
+    def is_valid_uuid(resourceId):
+        try:
+            uuid.UUID(str(resourceId))
+        except ValueError:
+            raise ValueError("Resource Id should be a valid GUID")
+
+    def resource_history_query(self, resourceId: str):
+        logging.debug("Validate sainty of resourceId")
```
[the remark this answers is not in the document]
Comment (anatbal) [reply api_app/db/repositories/resources_history.py]: fixed :)

File: api_app/db/repositories/resources_history.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            resourceId=resource.id,
+            isEnabled=resource.isEnabled,
+            properties=resource.properties,
+            resourceVersion=resource.resourceVersion,
+            updatedWhen=resource.updatedWhen,
+            user=resource.user,
+            templateVersion=resource.templateVersion
+        )
+        # Check if already save this resourceVersion in history container
+        existing_resource_history_item = await self.query(self.resource_history_with_resource_version_query(resource.id, resource.resourceVersion))
```
[the remark this answers is not in the document]
Comment (anatbal) [reply api_app/db/repositories/resources_history.py]: Removed

**Comment by anatbal** [comment]
https://github.com/microsoft/AzureTRE/actions/runs/3896064292 ran successfully. force approving

**Pull request merged**